break_glass/program.py

```diff
@@ -0,0 +1,119 @@
+slack = slack_client("slack_connection")
+
+
+def on_slack_slash_command(event):
+    """Sends a form to request approval for a ticket."""
+    trigger_id = event.data["trigger_id"]
+    request_modal = Path("request_modal.json.txt").read_text()
+    slack.views_open(trigger_id=trigger_id, view=request_modal)
+
+
+@autokitteh.activity
+def on_form_submit(event):
+    reason, issue_key, base_url, requester_id, requester_name = parse_event_data(event)
+
+    if not check_issue_exists(issue_key):
+        message = f"Ticket {issue_key} does not exist. Please try again."
+        slack.chat_postMessage(channel=requester_id, text=message)
+        return
+
+    email = slack.users_info(user=requester_id)["user"]["profile"]["email"]
+    if not validate_requester(issue_key, email):
+        message = f"You are not the assignee for ticket {issue_key}. Please try again."
+        slack.chat_postMessage(channel=requester_id, text=message)
+        return
+
+    send_approval_request(reason, issue_key, base_url, requester_id, requester_name)
+    slack.chat_postMessage(channel=requester_id, text="Request sent for approval.")
+
+
+@autokitteh.activity
+def on_approve_deny(event):
+    action_id = event.data["actions"][0]["action_id"]
+    _, requester, issue_key = action_id.split(" ")
+    approver_id = event.data["user"]["id"]
+    approver_info = slack.users_info(user=approver_id)
+
+    if event.data["actions"][0]["value"] == "Approve":
+        approver_email = approver_info["user"]["profile"]["email"]
+        jira.issue_add_comment(issue_key, f"Request approved by: {approver_email}")
+        message = f"Request approved by: <@{approver_info['user']['name']}>"
+        slack.chat_postMessage(channel=requester, text=message)
+    else:
+        print(f"Requester: {requester}")
+        message = f"Request denied by: <@{approver_info["user"]["name"]}>"
+        slack.chat_postMessage(channel=requester, text=message)
+
+
+def send_approval_request(reason, issue_key, base_url, requester_id, requester_name):
+    blocks = Path("approval_message.json.txt").read_text()
+    changes = [
+        ("Title", "Break Glass Request"),
+        ("RequestFromMessage", f"*Request from*: <@{requester_name}>"),
+        ("Ticket", f"*Ticket*: <{base_url}/browse/{issue_key}|{issue_key}>"),
+        ("Reason", "*Reason for request*: " + reason),
+        ("RequesterId", requester_id),
+        ("IssueKey", issue_key),
+    ]
+    for old, new in changes:
+        blocks = blocks.replace(old, new)
+    slack.chat_postMessage(channel=APPROVER, blocks=blocks)
+
+
+def parse_event_data(event):
+    form_data = event.data["view"]["state"]["values"]
+    reason = form_data["block_reason"]["reason"]["value"]
+    issue_key = form_data["block_issue_key"]["issue_key"]["value"]
+    base_url = os.getenv("jira_connection__AccessURL")
+    requester_id = event.data["user"]["id"]
+    requester_name = slack.users_info(user=requester_id)["user"]["name"]
```

Comment:
No need for this call - you can just return the ID, and `<@U1234>` would still be rendered by Slack as the user's display name (see https://api.slack.com/reference/surfaces/formatting#mentioning-users).

The ID is also guaranteed to remain unique forever, whereas the username is guaranteed to remain unique only while the referenced user exists and isn't replaced by a different one.